### DISCUSSION TITLE
Can't configure Asustor NAS App store version

### DISCUSSION THREAD

**Discussion by jab8778** [photoprism/photoprism#4620] [Q&A]
I have been using the Asustor App store version of PhotoPrism for the last two years, this is the first one I can't get to work after making even just one change to the ENV variables. I use Portainer CE and make simple configuration changes like PHOTOPRISM_FFMPEG_SIZE=7680. Once I redeploy the container it starts but can't see the MariaDB. I am a novice, that is why I use PortainerCE.

I have searched the forum and  troubleshooting docs, but unfortunately I don't know enough about Docker/PortainerCE to be able to figure out why the database link is broken

PhotoPrism® Essentials
Build 240915-e1280b2fb

PhotoPrismD
        bridge  172.XX.X.X      172.XX.X.X      02:42:xx:XX:XX:XX
[2024-10-29T16:34:50-04:00] config: waiting for the database to become available 
ERRO[2024-10-29T16:35:50-04:00] dial tcp: lookup mariadb on 192.XXX.XX.X.XX: no such host 
/opt/photoprism/bin/photoprism start

mariadb_mysql
               bridge   172.XX.X.X      172.XX.X.X1      02:42:xx:XX:XX:XX
2024-10-29 16:22:06 5 [Warning] Aborted connection 5 to db: 'photoprismdb' user: 'photoprism' host: '172.17.0.3' (Got an error reading communication packet

Any help or suggestions on where to turn would be greatly appreciated.

**Comment by omerdduran** [comment]
Hello! Taking a backup first might be a good idea.

Here's how to access the terminal and take a backup
https://docs.photoprism.app/getting-started/portainer/#command-line-interface

Settings you might want to check
1. For MariaDB:
```
MARIADB_DATABASE=photoprism
MARIADB_USER=photoprism
MARIADB_PASSWORD=insecure # Change this
MARIADB_ROOT_PASSWORD=insecure # Change this too
```

2. For PhotoPrism:
```
PHOTOPRISM_DATABASE_DRIVER=mysql
PHOTOPRISM_DATABASE_SERVER=mariadb:3306
PHOTOPRISM_DATABASE_NAME=photoprism
PHOTOPRISM_DATABASE_USER=photoprism
PHOTOPRISM_DATABASE_PASSWORD=insecure # Must match MariaDB password
```


Also make sure that the name of the MariaDB container is exactly “mariadb” and make sure that both containers are on the same network (Network section in Portainer). I hope this helps!

**Reply by jab8778** [reply]
Thank you for the suggestion. All variables are as noted and both containers are in the same (and only) Bridge Network. The fact that I have done this several times as new updates and preview builds have been released over the last two years, is what is so confusing. I have deleted the apps, including Portainer, even the app folders on the NAS and re-installed just to have the same issue.
